src/shared_key.rs

```diff
@@ -22,9 +23,85 @@ pub enum SharedKeyError {
     Generate,
 }
 
+const DITTO_IDENTITY_TAG: &str = "DITTO IDENTITY";
+
+#[derive(Debug, Serialize, Deserialize, PartialEq, Eq)]
+struct X509Auth {
+    certificates: Vec<Vec<u8>>,
+    private_key: Vec<u8>,
+}
+
+#[derive(Debug, Serialize, Deserialize, PartialEq, Eq)]
+enum JwtAuth {
+    AuthUrl { url: String },
+    AuthProvider { name: String },
+    Token { token: String },
+}
+
+#[derive(Debug, Serialize, Deserialize, PartialEq, Eq)]
+struct ManualIdentity {
+    version: u16,
+    private_key: Vec<u8>,
+    expiry: chrono::DateTime<chrono::Utc>,
+    identity_data: Vec<u8>,
+    inband_ca_pubkey_keys: Vec<Vec<u8>>,
+    // inband_certificate: InbandCertificate,
+    // inband_metadata: Option<??>,
+    x509_auth: Option<X509Auth>,
+    jwt: Option<JwtAuth>,
+}
+
+impl ManualIdentity {
+    pub fn new() -> Self {
+        Self {
+            version: 1,
+            private_key: vec![],
+            expiry: chrono::Utc::now(),
+            identity_data: vec![],
+            inband_ca_pubkey_keys: vec![],
+            x509_auth: None,
+            jwt: None,
+        }
+    }
+
+    pub fn to_string(&self) -> String {
+        let cbor = serde_cbor::to_vec(&self).unwrap();
+        let pem = pem::Pem {
```

Comment:
I propose we use the standard PEM format with our own tag. That way we get support for free in form of programming libraries & familiarity from developers. Compared to the previous proposal, we only need to move the `version` field into the base64 content.

Reply:
Sounds great 👍